src/main/java/org/folio/holdingsiq/model/OkapiData.java

```diff
@@ -2,34 +2,31 @@
 
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.HashMap;
 import java.util.Map;
 
 import lombok.Getter;
+import org.apache.commons.collections4.map.CaseInsensitiveMap;
+import org.folio.okapi.common.XOkapiHeaders;
 
 @Getter
 public class OkapiData {
 
-  private static final String OKAPI_TOKEN_HEADER = "x-okapi-token";
-  private static final String OKAPI_URL_HEADER = "x-okapi-url";
-  private static final String OKAPI_TENANT_HEADER = "x-okapi-tenant";
-
   private final Map<String, String> headers;
   private final String apiToken;
   private final String tenant;
   private final String okapiHost;
   private final int okapiPort;
   private final String okapiUrl;
+  private final String userId;
 
   public OkapiData(Map<String, String> headers) {
-    this.headers = new HashMap<>(headers);
-
-    Map<String, String> lowercaseHeaders = getLowercaseHeaders(headers);
+    this.headers = new CaseInsensitiveMap<>(headers);
 
     try {
-      apiToken = lowercaseHeaders.get(OKAPI_TOKEN_HEADER);
-      tenant = lowercaseHeaders.get(OKAPI_TENANT_HEADER);
-      okapiUrl = lowercaseHeaders.get(OKAPI_URL_HEADER);
+      apiToken = this.headers.get(XOkapiHeaders.TOKEN);
+      tenant = this.headers.get(XOkapiHeaders.TENANT);
+      userId = this.headers.get(XOkapiHeaders.USER_ID);
```

Comment:
Should we add parsing userId from okapiToken as a backup in case userId header is missing?

Reply:
I don't think so. It was a bad approach to parsing the token.